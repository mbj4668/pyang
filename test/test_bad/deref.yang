module deref {
  yang-version 1.1;
  namespace urn:deref;
  prefix d;
  
  list foo {
    key "a";
    leaf a {
      type string;
    }
    container b {
      leaf c {
        type string;
      }
    }
  }

  leaf bar {
    type leafref {
      path "../foo/a";
    }
  }

  leaf b1 {
    type string;
    must "deref(../bar)/../c";  // c not found
  }
  
  leaf b2 {
    type string;
    must "deref(../bax)/../c";  // bax not found
  }
  
  leaf b3 {
    type string;
    must "deref(.)";  // . not leafref
  }
}

