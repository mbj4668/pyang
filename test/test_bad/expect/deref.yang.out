deref.yang:26: warning: XPATH_NODE_NOT_FOUND1
deref.yang:31: warning: XPATH_NODE_NOT_FOUND2
deref.yang:36: warning: XPATH_DEREF_TARGET
